pkg/scheduler/objects/queue.go

```diff
@@ -691,11 +700,14 @@ func (sq *Queue) RemoveApplication(app *Application) {
 	sq.Lock()
 	delete(sq.applications, appID)
 	delete(sq.appPriorities, appID)
+	delete(sq.appPreemptionPriorities, appID)
 	delete(sq.allocatingAcceptedApps, appID)
 	priority := sq.recalculatePriority()
+	preemptionPriority := sq.recalculatePreemptionPriority()
 	sq.Unlock()
```

Comment:
Can we refactor all these calls into a separate locked function `internalRemoveApp(appID string) int32` makes it much cleaner and a defer can be used to guarantee the unlock even if we add things later.

Reply:
Done